library/core/src/ptr/const_ptr.rs

```diff
@@ -792,11 +795,11 @@ impl<T: ?Sized> *const T {
     #[unstable(feature = "const_raw_ptr_comparison", issue = "53020")]
     #[rustc_const_unstable(feature = "const_raw_ptr_comparison", issue = "53020")]
     #[inline]
-    pub const fn guaranteed_eq(self, other: *const T) -> bool
+    pub const fn guaranteed_eq(self, other: *const T) -> Option<bool>
     where
         T: Sized,
     {
-        intrinsics::ptr_guaranteed_eq(self, other)
+        (self as *mut T).guaranteed_eq(other as _)
```

Comment:
Uhh, I would personally prefer the implementation to go the other way around (mut ptr calls const_ptr impl).